src/ubuntu_core_kiosk_main.cpp

```diff
@@ -20,22 +20,56 @@
 
 #include <miral/command_line_option.h>
 #include <miral/display_configuration.h>
+#include <miral/external_client.h>
 #include <miral/keymap.h>
 #include <miral/runner.h>
 #include <miral/set_window_management_policy.h>
+#include <miral/wayland_extensions.h>
 
 int main(int argc, char const* argv[])
 {
     using namespace miral;
+    using namespace std::string_literals;
+
     MirRunner runner{argc, argv};
 
     DisplayConfiguration display_config{runner};
- 
+    WaylandExtensions wayland_extensions;
+
+    for (auto const& extension : {
+        WaylandExtensions::zwlr_layer_shell_v1, // For swaybg
+        "zwp_pointer_constraints_v1",           // Useful for games
+        "zwp_relative_pointer_manager_v1"})     // Useful for games
+    {
+        wayland_extensions.enable(extension);
+    }
+
+    ExternalClientLauncher launcher;
+
+    auto background = "#1f1f1f"s;
+
+    auto startup_background = [&](std::string const& swaybg_launch_path)
+        {
+            launcher.launch({swaybg_launch_path, "-c", background});
+        };
+
+    auto const default_swaybg_launch_path = []
```

Comment:
Does this need to be a lambda? It seems it's being executed unconditionally just below. Why not do the logic here and set a variable with the result?

Reply:
Fixed

Reply:
I'm not crazy about using lamdas for control flow. Discussion in matermost.

Reply:
In this case, right now, I don't think the immediately-executed-lambda is particularly valuable; it is immediately used and then just as immediately falls out of scope, so the benefit of `const`ing the variable is limited.

I don't *object* to using an IEL, though.

Reply:
Fixed more